main.py

```diff
@@ -48,14 +48,30 @@ def wrap_text(message, max_width, indent=24):
 
     return wrapped_message
 
+def chk_username_presence(mesg_json):
+    new_name = mesg_json.split(sepr)[1]
+    chatroom_id = mesg_json.split(sepr)[2]
+    if new_name in getallus(chatroom_id):
+        return True
+    else:
+        return False
+            
 
 async def chatroom(websocket, path):
     if not websocket in USERS:
         USERS[websocket] = ""
     try:
         async for mesgjson in websocket:
             if sepr in mesgjson and websocket in USERS:
-                if USERS[websocket] == "":
+                if (mesgjson.split(sepr)[0] == "CHKUSR") & (len(mesgjson.split(sepr)) == 3) :
+                    result = str(chk_username_presence(mesgjson))
+                    await websocket.send(result)
+                    if(result == "True"):
+                        await websocket.close()
+                        USERS.pop(websocket)
+                        return
```

Comment:
Does this return serve any practical function?

Reply:
Returning will remover the listening instance for that specific user's websocket who couldn't join if I'm not wrong. Please correct me if I'm wrong 😅